src/inatinqperf/adaptors/milvus_adaptor.py

```diff
@@ -0,0 +1,164 @@
+    HNSW = "HNSW"
+    HNSW_SQ = "HNSW_SQ"
+    HNSW_PQ = "HNSW_PQ"
+
+    @classmethod
+    def _missing_(cls, value: str) -> "MilvusIndexType | None":
+        value = value.lower()
+        for member in cls:
+            if member.value.lower() == value:
+                return member
+        return None
+
+
+class Milvus(VectorDatabase):
+    """Milvus vector database."""
+
+    DATABASE_NAME: str = "default"
+    COLLECTION_NAME: str = "collection_name"
+    INDEX_NAME: str = f"{COLLECTION_NAME}_index"
+
+    @logger.catch
+    def __init__(
+        self,
+        dataset: HuggingFaceDataset,
+        metric: Metric,
+        index_type: MilvusIndexType | str,
+        index_params: dict | None = None,
+        host: str = "localhost",
+        port: str = "19530",
+    ) -> None:
+        super().__init__(dataset, metric)
+        self.host = host
+        self.port = port
+        self.index_type = MilvusIndexType(index_type)
+        try:
+            connections.connect(host="localhost", port="19530")
+            server_type = utility.get_server_type()
+            logger.add(f"Milvus server is running. Server type: {server_type}")
+        except Exception:
+            logger.exception("Milvus server is not running or connection failed")
+
+        self.client = MilvusClient(uri=f"http://{self.host}:{self.port}")
+        self.metric = self._translate_metric(metric)
+
+        # Remove collection if it already exists
+        if self.client.has_collection(self.COLLECTION_NAME):
+            self.client.drop_collection(self.COLLECTION_NAME)
+
+        # Define collection schema
+        self.schema = self.client.create_schema(
+            auto_id=False,
+            enable_dynamic_schema=True,
+        )
+        self.schema.add_field(field_name="id", datatype=DataType.INT64, is_primary=True)
+        self.schema.add_field(field_name="vector", datatype=DataType.FLOAT_VECTOR, dim=self.dim)
+
+        self.index_params = self.client.prepare_index_params()
+        self.index_params.add_index(
+            field_name="vector",
+            index_type=self.index_type.value,
+            index_name=self.INDEX_NAME,
+            metric_type=self.metric,
+            params=index_params if index_params else {},
+        )
+
+        self.client.create_collection(
+            collection_name=self.COLLECTION_NAME,
+            schema=self.schema,
+            index_params=self.index_params,
+        )
+
+        batch_size = 1000
+
+        for i in tqdm(range(0, len(dataset), batch_size)):
+            batch_data = []
+            end = min(i + batch_size, len(dataset))
+            for k in range(i, end):
+                rid = int(dataset[k]["id"])
+                vec = dataset[k]["embedding"]
+                batch_data.append({"id": rid, "vector": vec})
+            self.client.insert(collection_name=self.COLLECTION_NAME, data=batch_data)
+
+        self.client.load_collection(collection_name=self.COLLECTION_NAME)
```

Comment:
Maybe add a comment explaining what this does? Does it load the collection to RAM?